docs/src/literate-tutorials/dg_heat_equation.jl

```diff
@@ -0,0 +1,397 @@
+#     trial and test functions in the quadrature point ``\textbf{x}_q``. However, to
+#     underline the strong parallel between the weak form and the implementation, this
+#     example uses the symbols appearing in the weak form.
+
+function assemble_element!(Ke::Matrix, fe::Vector, cellvalues::CellValues)
+    n_basefuncs = getnbasefunctions(cellvalues)
+    ## Reset to 0
+    fill!(Ke, 0)
+    fill!(fe, 0)
+    ## Loop over quadrature points
+    for q_point in 1:getnquadpoints(cellvalues)
+        ## Get the quadrature weight
+        dΩ = getdetJdV(cellvalues, q_point)
+        ## Loop over test shape functions
+        for i in 1:n_basefuncs
+            δu  = shape_value(cellvalues, q_point, i)
+            ∇δu = shape_gradient(cellvalues, q_point, i)
+            ## Add contribution to fe
+            fe[i] += δu * dΩ
+            ## Loop over trial shape functions
+            for j in 1:n_basefuncs
+                ∇u = shape_gradient(cellvalues, q_point, j)
+                ## Add contribution to Ke
+                Ke[i, j] += (∇δu ⋅ ∇u) * dΩ
+            end
+        end
+    end
+    return Ke, fe
+end
+
+function assemble_interface!(Ki::Matrix, iv::InterfaceValues, μ::Float64)
+    ## Reset to 0
+    fill!(Ki, 0)
+    ## Loop over quadrature points
+    for q_point in 1:getnquadpoints(iv)
+        ## Get the normal to face A
+        normal = getnormal(iv, q_point)
+        ## Get the quadrature weight
+        dΓ = getdetJdV(iv, q_point)
+        ## Loop over test shape functions
+        for i in 1:getnbasefunctions(iv)
+            ## Multiply the jump by the normal, as the definition used in Ferrite doesn't include the normals.
+            δu_jump = shape_value_jump(iv, q_point, i) * normal
+            ∇δu_avg = shape_gradient_average(iv, q_point, i)
+            ## Loop over trial shape functions
+            for j in 1:getnbasefunctions(iv)
+                ## Multiply the jump by the normal, as the definition used in Ferrite doesn't include the normals.
+                u_jump = shape_value_jump(iv, q_point, j) * normal
+                ∇u_avg = shape_gradient_average(iv, q_point, j)
+                ## Add contribution to Ki          
+                Ki[i, j] += -(δu_jump ⋅ ∇u_avg + ∇δu_avg ⋅ u_jump)*dΓ +  μ * (δu_jump ⋅ u_jump) * dΓ
+            end
+        end
+    end
+    return Ki
+end
+
+function assemble_boundary!(fe::Vector, fv::FaceValues)
+    ## Reset to 0
+    fill!(fe, 0)
+    ## Loop over quadrature points
+    for q_point in 1:getnquadpoints(fv)
+        ## Get the normal to face A
+        normal = getnormal(fv, q_point)
+        ## Get the quadrature weight
+        ∂Ω = getdetJdV(fv, q_point)
+        ## Loop over test shape functions
+        for i in 1:getnbasefunctions(fv)
+            ## Multiply the jump by the normal, as the definition used in Ferrite doesn't include the normals.
+            δu = shape_value(fv, q_point, i)
+            fe[i] = normal[2] * δu * ∂Ω
```

Comment:
With the boundary condition `n cdot nabla u = n[2]` (although not explicitly stated, only via the conditional in the equation), I found it a bit hard to "see through" the implementation here. Could something like the following be easier to read?
(Was the comment a left-over from the internal boundary contribution?)
```suggestion
            δu = shape_value(fv, q_point, i)
            boundary_flux = normal[2]
            fe[i] = boundary_flux * δu * ∂Ω
```